internal/api/poller.go

```diff
@@ -70,10 +90,17 @@ func (ip *InterruptiblePoller) Run(ctx context.Context, ch <-chan struct{}, acti
 			case <-j.ctx.Done():
 				// Job's context was canceled. Nothing else to do here.
 			case <-j.run():
-				if j.ctx.Err() == nil {
+				backoffDuration := resetDuration
+				if ip.isBackoffError(jt.logger, j.err) {
+					backoffDuration = ip.backoff.Step()
+					jt.logger.Debugf("poller backing off, next run %s)", backoffDuration)
+				}
+				if backoffDuration > resetDuration {
+					<-time.After(backoffDuration)
```

Comment:
this will likely end up being a select so that the wait can be interupted by the worker ranging over the `ch` passed in to run